pennylane/operation.py

```diff
@@ -961,19 +961,6 @@ def kraus_matrices(self):
         """
         return self._kraus_matrices(*self.parameters)
 
-    def __init__(self, *params, wires=None, do_queue=True):
-
-        # check parameters are valid
-        if self.par_domain == "R" and any(not 0 <= np.real(p) <= 1 for p in params):
-            raise ValueError("Channel probability parameters should be numbers between 0 and 1.")
-
-        # check the grad_method validity
-        if self.par_domain == "R" and self.grad_method not in (None, "F"):
-            raise ValueError("Analytic gradients can not be used for quantum channels.")
```

Comment:
I've tried an example using both `AmplitudeDamping` and `PhaseFlip`, and it computes the gradient no problem. Is that because the inclusion of `AmplitudeDamping` forces it to use non-analytic gradient method?

Reply:
Prior to this PR, all channel gradients were computed using finite difference. If you were testing from master, then indeed everything should work. If you were testing from this branch, then computing a gradient of `PhaseFlip` without removing this check would throw an error, since `grad_method` is now set to `'A'` for this channel